src/main/java/Deadline.java

```diff
@@ -0,0 +1,59 @@
+import java.time.LocalDateTime;
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDateTime byDeadline;
+
+    /**
+     * Creates a deadline event.
+     * @param description task description
+     * @param byDeadline deadline for task
+     * @throws DukeException when there is a problem parsing the date time string provided by user
+     */
+    public Deadline(String description, String byDeadline) throws DukeException {

```

Comment:
Over here, Deadline takes in a string of deadline. Would it be alright to take in a LocalDateTime instead, as the parsing role should be Parser's?